### PR TITLE
Add API for unexecuted op

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43743 Update aliasing in tensorexpr fuser
* #43742 use types in the IR instead of vmap
* #43636 Add inline call to profiling executor pipeline
* #43635 Use prim::TensorExprGroup interned symbol
* #43634 Update requires grad property
* #43633 Specialize optionals for grad_sum_to_size
* #43632 Add undefined specializations in backward
* #43631 refactor tests
* #43630 Refactor pass to class
* **#43629 Add API for unexecuted op**

We have a few places where we count the size a block / subgraph - it's nice to have a shared API to ignore operators that are not executed in the optimized graph (will be used when i add a new profiling node in PR ^^)

Differential Revision: [D23358807](https://our.internmc.facebook.com/intern/diff/D23358807)

### CODE REVIEW

**Review by ZolotukhinM — APPROVED** [review]
👍

File: torch/csrc/jit/ir/ir.h [hunk 1]
```diff
@@ -435,6 +435,11 @@ struct TORCH_API Node {
   bool isNondeterministic() const;
   bool hasSideEffects() const;
 
+  // instructions lowered by the interpreter and not run in the optimized graph
+  bool notExecutedOp() const {
+    return kind_ == prim::Constant || kind_ == prim::profile;
```
Comment (Krovatkin) [inline torch/csrc/jit/ir/ir.h:440]: nitpick: do we add `prim::profile_optional` later on?

**Review by Krovatkin — APPROVED** [review]
:shipit:

File: torch/csrc/jit/ir/ir.h (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/ir/ir.h:440]: yea, we do (although maybe we hadnt at the point of your reviewing)

**Pull request closed without merging**